dts/arm/nxp/nxp_imx7d_m4.dtsi

```diff
@@ -56,9 +56,10 @@
 			reg = <0x20200000 DT_SIZE_K(128)>;
 		};
 
-		ocram_s_code: code@20180000 {
+		/* OCRAM_S 0x20180000 is aliased at 0 */
+		ocram_s_code: code@0 {
 			compatible = "nxp,imx-code-bus";
-			reg = <0x20180000 DT_SIZE_K(32)>;
+			reg = <0x00000000 DT_SIZE_K(32)>;
```

Comment:
@jlh-makeen , revert this update?

Reply:
I can explain this choice:

The Linux imx_rproc driver is NOT mapping the 0x20180000 address space to 0x00180000, which means that when user selects zephyr,flash=&ocram_s_code, rproc doesn't load the irq_vector and code at the right memory space. 
Even though the M4 itself understands 0x20180000 which is the CODE bus where instructions are placed, rproc will not load the elf-file correctly. 
This is because OCRAM_S is at address 0x00180000 on the A7 side.

By using address 0, we are guaranteed that all CODE addresses/pointers will be aliased to the OCRAM_S_CODE address by the M4 and to 0x00180000 by the imx_rproc driver.

That reasoning is valid for imx7d, imx6sx and other soc like imx8

https://github.com/nxp-imx/linux-imx/blob/lf-6.6.y/drivers/remoteproc/imx_rproc.c#L262

Unless I am proven wrong and explained why using the aliased address instead of the raw address here is not a good idea, I'd rather keep this changeset :)

Reply:
Yes, indeed, actually we also need to add 0x20180000 mapping in imx_rproc driver as well, but currently both 0x00000000 and 0x00180000 are okay at here.